sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -211,50 +216,44 @@ public void AddTag(string name, object value)
                 }
             }
 
-            private List<ActivityLink>? GetActivitySourceLinkCollection()
+            private IEnumerable<Activity> GetDiagnosticSourceLinkCollection()
```

Comment:
I wouldn't recommend generalizing the return type to `IEnumerable<T>` unless we have to.   Any chance we can use `IList<T>` or `IReadOnlyList<T>`?

Reply:
why not? It's passed over as `IEnumerable`

```csharp
_currentActivity = new DiagnosticActivity(_activityName)
{
    Links = GetDiagnosticSourceLinkCollection(),
};
```
So if I return something else, I'd need to cast it.

Reply:
If that's the form we get it in, agree there's no value.   I thought I saw this being created as a `List` instance below and we were down-casting the return.

Reply:
makes sense, changed to `IReadOnlyList`